src/test/TUSDCurveCallee.t.sol

```diff
@@ -0,0 +1,332 @@
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(minProfit)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertGe(Token(dai).balanceOf(address(123)), minProfit);
+    }
+
+    function test_profitBigAmt() public {
+        uint256 minProfit = 1000 * WAD;
+        uint256 amt = 10_000_000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(minProfit)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertGe(Token(dai).balanceOf(address(123)), minProfit);
+    }
+
+    function test_profitAllCollateral() public {
+        uint256 minProfit = 1000 * WAD;
+        uint256 amt = 26_000_000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(minProfit)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertGe(Token(dai).balanceOf(address(123)), minProfit);
+    }
+
+    function test_maxPrice() public {
+        uint256 amt = 20000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(this),
+            address(gemJoin),
+            uint256(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        address osm = Chainlog(chainlog).getAddress("PIP_TUSD");
+        Hevm(hevm).store({
+            c:   osm,
+            loc: keccak256(abi.encode(address(this), uint256(0))),
+            val: bytes32(uint256(1))
+        });
+        uint256 max = uint256(Osm(osm).read()) * 1e9; // WAD * 1e9 = RAY
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  max,
+            who:  address(callee),
+            data: data
+        });
+    }
+
+    function testFail_maxPrice() public {
+        uint256 amt = 20000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(this),
+            address(gemJoin),
+            uint256(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 5);
```

Comment:
What is the reasoning behind the choice of this specific number ? Maybe add a comment to explain.

Reply:
No real reasoning as this was ported from the test of WstETH.
Added the following comment:
"Make sure price goes down enough to make up for the Curve slippage'